semester_1/kr2.1/task2/list/list.c

```diff
@@ -0,0 +1,132 @@
+#include <stdlib.h>
+#include <stdio.h>
+
+#include "list.h"
+
+typedef struct Node
+{
+    int value;
+    struct Node *next;
+} Node;
+
+typedef struct List
+{
+    Node *head;
+} List;
+
+void printList(List const *const list)
+{
+    printf("\n[");
+    for (Node *currentNode = list->head; currentNode != NULL; currentNode = currentNode->next)
+    {
+        printf("%d", currentNode->value);
+        if (currentNode->next != NULL)
+        {
+            printf(", ");
+        }
+    }
+    printf("]\n");
+}
+
+List *createList(void)
+{
+    List *newList = (List *)calloc(1, sizeof(List));
+
+    return newList;
```

Comment:
Тут короче без временной переменной:
```suggestion
    return (List *)calloc(1, sizeof(List));
```